### DISCUSSION TITLE
Suggests updating to dependency that is in beta with `ignoreUnstable` set to `true`

### DISCUSSION THREAD

**Discussion by martinfrancois** [renovatebot/renovate#26455] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

None

### Describe the problem

In my `docker-compose.yml` I have this image: https://hub.docker.com/_/traefik defined with the tag "v2.10". `ignoreUnstable` is set to `true`

mend created a PR suggesting to update it to "v3", however, there are only beta versions of traefik v3 available, so I would expect it to **not** suggest updating to "v3".

My guess is it suggests that as there is a tag available that is "v3", which it considers to be newer. One way to fix this would be when it suspects there is a new version, it could first check if any of the other tags with the same version that are not unstable releases. Another, probably more easy and accurate approach would be to compare which major version the "latest" tag points to. If "latest" as in this case points to a 2.x version, it should consider anything with a major version above 2 to be unstable.

A minimal reproduction repository can be found here: https://github.com/martinfrancois/reproduction-bug-renovate

### Relevant debug logs

<details><summary>Logs</summary>

```

DEBUG: 1 flattened updates found: traefik
DEBUG: Returning 1 branch(es)
DEBUG: config.repoIsOnboarded=false
DEBUG: packageFiles with updates
{
  "baseBranch": "master"
  "config": {
    "docker-compose": [
      {
        "deps": [
          {
            "depName": "traefik",
            "currentValue": "v2.10",
            "replaceString": "traefik:v2.10",
            "autoReplaceStringTemplate": "{{depName}}{{#if newValue}}:{{newValue}}{{/if}}{{#if newDigest}}@{{newDigest}}{{/if}}",
            "datasource": "docker",
            "updates": [
              {
                "bucket": "major",
                "newVersion": "3.0",
                "newValue": "3.0",
                "releaseTimestamp": "2023-12-01T11:33:14.815Z",
                "newMajor": 3,
                "newMinor": 0,
                "updateType": "major",
                "branchName": "renovate/traefik-3.x"
              }
            ],
            "packageName": "traefik",
            "versioning": "docker",
            "warnings": [],
            "sourceUrl": "https://github.com/containous/traefik",
            "registryUrl": "https://index.docker.io",
            "currentVersion": "v2.10",
            "isSingleVersion": true,
            "fixedVersion": "v2.10"
          }
        ],
        "packageFile": "docker-compose.yml"
      }
    ]
  }
}

DEBUG: detectSemanticCommits()
DEBUG: semanticCommits: returning "disabled" from cache
DEBUG: ensureOnboardingPr()
DEBUG: getBranchPr(renovate/configure)
DEBUG: findPr(renovate/configure, undefined, open)
DEBUG: findPr(renovate/configure, undefined, closed)
DEBUG: Filling in onboarding PR template
DEBUG: getConfigDesc()
DEBUG: getDescriptionArray()
DEBUG: getScheduleDesc()
DEBUG: No schedule
DEBUG: Found description array with length:8
DEBUG: getPrList()
DEBUG: Creating onboarding PR
DEBUG: Creating PR
{
  "title": "Configure Renovate"
  "head": "martinfrancois:renovate/configure"
  "base": "master"
  "draft": false
}

DEBUG: PR created (branch="renovate/configure")
{
  "pr": 1
  "draft": false
}

DEBUG: Adding labels '' to #1
INFO: Onboarding PR created
{
  "pr": "Pull Request #1"
}

DEBUG: addParticipants(pr=1)
DEBUG: processRepo()
DEBUG: getPrCache()
DEBUG: ensureDependencyDashboard()
DEBUG: Repo is onboarding - skipping dependency dashboard
DEBUG: validateReconfigureBranch()
DEBUG: No reconfigure branch found
DEBUG: Removing any stale branches
DEBUG: config.repoIsOnboarded=false
DEBUG: Branch lists
{
  "branchList": [
    "renovate/configure"
  ]
  "renovateBranches": [
    "renovate/configure"
  ]
}

DEBUG: remainingBranches=
DEBUG: No branches to clean up
DEBUG: Cleaning up Renovate refs: refs/renovate/*
DEBUG: PackageFiles.clear() - Package files deleted
DEBUG: Branch summary
{
  "baseBranches": [
    {
      "branchName": "master",
      "sha": "48402d45308aaaeb7eb44fd8a00a35ab21e1d3b9"
    }
  ]
  "branches": []
  "defaultBranch": "master"
  "inactiveBranches": [
    "renovate/traefik-3.x"
  ]
}

DEBUG: branches info extended
{
  "branchesInformation": [
    {
      "branchName": "renovate/traefik-3.x",
      "prNo": null,
      "prTitle": "Update traefik Docker tag to v3",
      "upgrades": [
        {
          "datasource": "docker",
          "depName": "traefik",
          "displayPending": "",
          "fixedVersion": "v2.10",
          "currentVersion": "v2.10",
          "currentValue": "v2.10",
          "newValue": "3.0",
          "newVersion": "3.0",
          "packageFile": "docker-compose.yml",
          "updateType": "major",
          "packageName": "traefik"
        }
      ]
    }
  ]
}


```

</details>

### Have you created a minimal reproduction repository?

I have linked to a minimal reproduction in the description above

**Comment by rarkins** [comment, marked as answer]
v3 is a stable tag. It's a mistake to publish it with unstable code. 

It is not a renovate _problem_ so closing this discussion.

**Reply by martinfrancois** [reply]
@rarkins Thanks for getting back to me so quickly. I was just about to open an issue in the Traefik repository when I saw there is already one about this and the maintainers seems to disagree about this: https://github.com/traefik/traefik-library-image/issues/44

**Reply by rarkins** [reply]
It's a mistake. A tag like v3-beta would be appropriate until it's stable.

Semver does not define constraints itself so there is such concept as v3 or 3.0 in the spec.

In npm semver and probably most other implementations which use ranges, a constraint like 3 or ^3.0.0 does not match pre-release 3.0.0-x versions.

Ultimately a v3 tag with unstable code is user unfriendly, plain and simple
